### PR TITLE
Use TextBlock.TextWrapping in the completion tooltip

### LINKED ISSUE #718: Completion tooltip doesn't wrap and gets cut off at the edge of the monitor
Ported from TFS WorkItem: **945982**

------------------------------

**Repro Steps:**

**Affected Branch:** <BR>**Affected Build:** <BR>**Affected Product Language:** <BR><BR>**Steps to reproduce:**<BR><BR>1. If the completion list's tooltip is wide enough, it will reach the edge of the monitor, where it's then cut off. It would be much better if we could somehow wrap it.

------------------------------

**Revisions:**

1) Created By Ravi Chande (5/14/2014 2:10:51 PM)

------------------------------

2) Edited By Ravi Chande (5/15/2014 9:50:49 AM)

Added a screenshot.

------------------------------

3) Edited By Mark Wilson-Thomas (5/15/2014 1:19:11 PM)

[t] This is not a regression. It's a good suggestion, and one we should consider. Copy to backlog?

------------------------------

4) Edited By Mark Wilson-Thomas (7/25/2014 9:35:52 AM)

Hi Ravi

&nbsp;

This is a good suggestion. The behavior is not a regression, and we've not had any external customer comment on it, so we're copying to backlog pending customer comment rather than investing in a fix right now.

&nbsp;

Thanks

Mark (EdEx Triage)

------------------------------

5) Edited By Ravi Chande (11/14/2014 3:11:27 PM)

C++ and C# in Dev12 both have the correct wrapping behavior, so I think this is actually just a Roslyn bug. Reactivating. If we decide to take this, let's notify the editor so they can remove it from their backlog (since the work is on our side).

------------------------------

6) Edited By Kevin Pilch-Bisson (11/17/2014 9:12:13 AM)

We're not sure what in the API would allow for wrapping here.

------------------------------

7) Edited By Kevin Pilch-Bisson (11/17/2014 9:13:29 AM)

Is this something we do specially to get colors in the tooltip?

------------------------------

8) Edited By Kevin Pilch-Bisson (2/19/2015 2:25:46 PM)

Port to GitHub.

------------------------------

### PR DESCRIPTION
**Pull request by rchande**
Fixes #718.

cc @balajikris @basoundr @brettfo @davkean @dpoeschl @jasonmalinowski @Pilchie

### CODE REVIEW

File: src/EditorFeatures/Core/Shared/Extensions/SymbolDisplayPartExtensions.cs [hunk 1]
```diff
@@ -27,14 +28,14 @@ public static Run ToRun(this SymbolDisplayPart part, IClassificationFormatMap fo
             return run;
         }
 
-        public static TextBlock ToTextBlock(this ImmutableArray<SymbolDisplayPart> parts, ClassificationTypeMap typeMap)
+        public static TextBlock ToTextBlock(this ImmutableArray<SymbolDisplayPart> parts, ClassificationTypeMap typeMap, TextWrapping textWrapping = TextWrapping.NoWrap)
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Shared/Extensions/SymbolDisplayPartExtensions.cs]: I'd rather not have optional arguments on internal APIs.  Instead I'd rather the caller have to think about it.  There aren't very many callers of these, can you just update them and make this required?

Comment (rchande) [inline src/EditorFeatures/Core/Shared/Extensions/SymbolDisplayPartExtensions.cs]: Indeed the only other caller sets the resultant TextBlock to use TextWrapping.Wrap. I can probably remove the parameter always wrap the TextBlock.

**Comment by balajikris** [comment]
:+1:  for the changes. Is it possible to unit test this with a long doc comment?

**Comment by rchande** [comment]
@balajikris We should probably modify VerifyQuickInfoAction to assert that it finds the text inside a TextBlock with wrapping set.

**Comment by rchande** [comment]
@Pilchie @jasonmalinowski @davkean Could one of you please take a look for a senior-level sign off?

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**